### PR TITLE
EG-1201, UI code single language

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/models/ocelot/Phrase.scala [hunk 1]
```diff
@@ -18,16 +18,19 @@ package models.ocelot
 
 import play.api.libs.json.Reads._
 import play.api.libs.json._
+import play.api.i18n.Lang
 
-case class Phrase(langs: Vector[String])
+case class Phrase(english: String, welsh: String) {
```
Comment (woollett-r) [inline app/models/ocelot/Phrase.scala:23]: That's gonna hurt!

File: test/models/ocelot/PhraseSpec.scala [hunk 2]
```diff
@@ -62,9 +70,8 @@ class PhraseSpec extends BaseSpec with ProcessJson {
 
       val protoTypePhrases = Json.parse(prototypePhrasesSection).as[Vector[Phrase]]
       val thirdPhraseIndex = 2
-      val welshLangIndex = 1
 
-      protoTypePhrases(thirdPhraseIndex).langs(welshLangIndex) shouldBe "Welsh: Overview"
+      protoTypePhrases(thirdPhraseIndex).welsh shouldBe "Welsh: Overview"
```
Comment (woollett-r) [inline test/models/ocelot/PhraseSpec.scala:74]: For completeness I would test the English phrase as well.

File: test/models/ui/UIComponentsSpec.scala [hunk 3]
```diff
@@ -17,105 +17,55 @@
 package models.ui
 
 import base.{BaseSpec, TestConstants}
-import play.api.i18n.Lang
 
 class UIComponentsSpec extends BaseSpec with TestConstants {
-
   val h1English: String = "Heading level 1 text"
```
Comment (woollett-r) [inline test/models/ui/UIComponentsSpec.scala:22]: I don't think the Welsh variables, such as h1Welsh, are used anymore. If not please remove.

File: test/services/EnglishTextBuilderSpec.scala [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    val answerWithNoHint = Phrase("Yes", "Welsh, Yes")
+    val answerWithHint = Phrase("Yes[hint:You agree with the assertion]", "Welsh, Yes[hint:Welsh, You agree with the assertion]")
+
+    val answer = Text("Yes")
+    val hint = Text("You agree with the assertion")
+  }
+
+  "TextBuilder placeholder parsing" must {
+
+    "Convert label reference with default output format placeholders within phrase to LabelRef TextItems" in new Test {
```
Comment (woollett-r) [inline test/services/EnglishTextBuilderSpec.scala:57]: In a number of the tests below the English and Welsh text are identical. As such we cannot be certain that the TextBuilder class is processing the English text.

File: test/services/EnglishUIBuilderSpec.scala [hunk 5]
```diff
@@ -499,7 +472,7 @@ class UIBuilderSpec extends BaseSpec with ProcessJson {
       val p = uiBuilder.buildPage("/start", Seq(SubSectionCallout(headingPhrase, Seq.empty, false),
                                                 tableRowGroup))
       p.components match {
-        case Seq(Table(Text(headingPhrase,_), tableHeading, _)) => succeed
+        case Seq(Table(Text(headingPhrase), tableHeading, _)) => succeed
```
Comment (woollett-r) [inline test/services/EnglishUIBuilderSpec.scala:475]: I'm not certain how this is working. I can't see an apply method that takes a Phrase argument and there is no toString on Phrase.

File: test/base/BaseSpec.scala [hunk 6]
```diff
@@ -19,6 +19,15 @@ package base
...
 import org.scalatest.{Matchers, WordSpec}
 import play.api.libs.json._
+import play.api.i18n.Lang
+
+trait EnglishLanguage {
+  implicit val lang: Lang = Lang("en")
+}
+
+trait WelshLanguage {
+  implicit val lang: Lang = Lang("en")
```
Comment (woollett-r) [inline test/base/BaseSpec.scala:29]: This looks wrong.

File: test/services/EnglishUIBuilderSpec.scala [hunk 7]
```diff
@@ -999,7 +945,7 @@ class UIBuilderSpec extends BaseSpec with ProcessJson {
   }
 
   trait InputTest {
-
+    implicit val lang: Lang = Lang("en")
```
Comment (woollett-r) [inline test/services/EnglishUIBuilderSpec.scala:948]: Is this not defined in the English language trait?

File: test/services/WelshTextBuilderSpec.scala [hunk 8]
```diff
@@ -54,185 +48,106 @@ class TextBuilderSpec extends BaseSpec {
...
 
-    val answer = Text("Yes", "Welsh, Yes")
-    val hint = Text("You agree with the assertion", "Welsh, You agree with the assertion")
+    val answer = Text("Yes")
+    val hint = Text("You agree with the assertion")
   }
 
   "TextBuilder placeholder parsing" must {
 
     "Convert label reference with default output format placeholders within phrase to LabelRef TextItems" in new Test {
```
Comment (woollett-r) [inline test/services/WelshTextBuilderSpec.scala:57]: As with the English text builder some of these tests need the English and Welsh text to differ.

File: test/services/WelshTextBuilderSpec.scala [hunk 9]
```diff
@@ -54,185 +48,106 @@ class TextBuilderSpec extends BaseSpec {
...
-
-      txt.english(2) shouldBe lEnWords2
-      txt.welsh(2) shouldBe lCyWords2
-
-      txt.english(3).toWords shouldBe Link("https://www.gov.uk", link2EnWords).toWords
-      txt.welsh(3).toWords shouldBe Link("https://www.gov.uk", link2CyWords).toWords
-
-      txt.english(4) shouldBe lEnWords3
-      txt.welsh(4) shouldBe lCyWords3
+      txt.items(0) shouldBe words1
```
Comment (woollett-r) [inline test/services/WelshTextBuilderSpec.scala:95]: This test will fail when lang equals "cy"

File: test/services/WelshUIBuilderSpec.scala [hunk 10]
```diff
@@ -0,0 +1,1388 @@
...
+      val bulletPointListStanzas = Seq(
+        KeyedStanza("start", PageStanza("/blah", Seq("1"), false)),
+        KeyedStanza("1", instructionGroup)
+      )
+      val bulletPointListPage = Page(Process.StartStanzaId, "/blah", bulletPointListStanzas, Seq.empty)
+      val uiPage = uiBuilder.buildPage(bulletPointListPage.url, bulletPointListPage.stanzas.collect{case s: VisualStanza => s})
+
+      uiPage.components.length shouldBe 1
+
+      // Check contents of bullet point list
```
Comment (woollett-r) [inline test/services/WelshUIBuilderSpec.scala:682]: Text definitions below differ slightly.

File: test/services/WelshUIBuilderSpec.scala [hunk 11]
```diff
@@ -0,0 +1,1388 @@
...
+
+          b.listItems.size shouldBe 2
+
+          b.listItems.head shouldBe bulletPointOne
+          b.listItems.last shouldBe bulletPointTwo
+        case _ => fail("Did not find bullet point list")
+      }
+    }
+
+    "Process page with a simple instruction group from prototypeJson" in new Test {
```
Comment (woollett-r) [inline test/services/WelshUIBuilderSpec.scala:700]: These tests should fail when the correct language is used.

File: test/views/components/ConfirmationPanelSpec.scala [hunk 12]
```diff
@@ -55,20 +55,14 @@ class ConfirmationPanelSpec extends BaseSpec with ViewFns with GuiceOneAppPerSui
     val englishBodyText2: String = "Second line in body"
     val welshBodyText2: String = "Welsh, Second line in body"
```
Comment (woollett-r) [inline test/views/components/ConfirmationPanelSpec.scala:56]: Are the Welsh string definitions still required?

File: test/views/components/GovukFooterSpec.scala [hunk 13]
```diff
@@ -35,17 +35,11 @@ class GovukFooterSpec extends ViewSpec with ViewFns with GuiceOneAppPerSuite {
 
   trait Test {
     private def injector: Injector = app.injector
-
     def messagesApi: MessagesApi = injector.instanceOf[MessagesApi]
     val footerLinks = app.injector.instanceOf[FooterLinks]
     val footer = app.injector.instanceOf[views.html.components.govukFooter]
     implicit def messages: Messages = messagesApi.preferred(Seq(Lang("en")))
```
Comment (woollett-r) [inline test/views/components/GovukFooterSpec.scala:41]: How do the Welsh tests work as I can't see an implicit Welsh declaration anywhere?

File: test/views/components/H1HeadingSpec.scala [hunk 14]
```diff
@@ -45,7 +45,7 @@ class H1HeadingSpec extends ViewSpec with GuiceOneAppPerSuite {
     val h1English: String = "Level 1 heading text"
     val h1Welsh: String = "Welsh Level 1 heading text"
```
Comment (woollett-r) [inline test/views/components/H1HeadingSpec.scala:46]: h1Welsh is no longer used.

File: test/views/components/H1HeadingWithFieldsetSpec.scala [hunk 15]
```diff
@@ -43,7 +43,7 @@ class H1HeadingWithFieldsetSpec extends ViewSpec with GuiceOneAppPerSuite {
     val h1English: String = "Level 1 heading text"
     val h1Welsh: String = "Welsh Level 1 heading text"
```
Comment (woollett-r) [inline test/views/components/H1HeadingWithFieldsetSpec.scala:44]: Again not used.

File: test/views/components/H1HeadingWithLabelSpec.scala [hunk 16]
```diff
@@ -40,7 +40,7 @@ class H1HeadingWithLabelSpec extends ViewSpec with GuiceOneAppPerSuite {
     val h1English: String = "Level 1 heading text"
     val h1Welsh: String = "Welsh Level 1 heading text"
```
Comment (woollett-r) [inline test/views/components/H1HeadingWithLabelSpec.scala:41]: Again not required

File: test/views/components/H3HeadingSpec.scala [hunk 17]
```diff
@@ -37,14 +37,11 @@ class H3HeadingSpec extends ViewSpec with GuiceOneAppPerSuite {
     implicit val labels: Labels = LabelCache()
     private def injector: Injector = app.injector
     def messagesApi: MessagesApi = injector.instanceOf[MessagesApi]
-
     val fakeRequest = FakeRequest("GET", "/")
     implicit def messages: Messages = messagesApi.preferred(fakeRequest)
-
     val h3English: String = "Level 3 heading text"
     val h3Welsh: String = "Welsh Level 3 heading text"
```
Comment (woollett-r) [inline test/views/components/H3HeadingSpec.scala:43]: This probably isn't used anymore.

File: test/views/components/InputSpec.scala [hunk 18]
```diff
@@ -121,7 +117,7 @@ class InputSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
       private val firstHint = hints.head
       private val hint1Attrs = elementAttrs(firstHint)
       hint1Attrs("class") shouldBe "govuk-hint"
-      firstHint.text() shouldBe Text(i1Hint).value(messages.lang).head.toString
+      firstHint.text() shouldBe Text(i1Hint).items.head.toString
```
Comment (woollett-r) [inline test/views/components/InputSpec.scala:120]: Is Text(i1Hint).items.head.toString a long winded way of saying i1Hint?

File: test/views/components/InputSpec.scala [hunk 19]
```diff
@@ -220,7 +216,7 @@ class InputSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
...
```
Comment (woollett-r) [inline test/views/components/InputSpec.scala:219]: As above

File: app/models/ocelot/Phrase.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/models/ocelot/Phrase.scala:23]: 👍

File: test/models/ocelot/PhraseSpec.scala (hunk 2 above)
Comment (cmrhmrc) [reply test/models/ocelot/PhraseSpec.scala:74]: added

File: test/models/ui/UIComponentsSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/models/ui/UIComponentsSpec.scala:22]: removed

File: test/services/EnglishTextBuilderSpec.scala (hunk 4 above)
Comment (cmrhmrc) [reply test/services/EnglishTextBuilderSpec.scala:57]: updated

File: test/models/ui/UIComponentsSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/models/ui/UIComponentsSpec.scala:22]: removed

File: test/services/EnglishUIBuilderSpec.scala (hunk 5 above)
Comment (cmrhmrc) [reply test/services/EnglishUIBuilderSpec.scala:475]: its a bit deceptive, the headingPhrase in the match (Table(Text(headingPhrase) ... is a filled in arg to the match not the val defined earlier. recoded to be less deceptive and test further

File: test/base/BaseSpec.scala (hunk 6 above)
Comment (cmrhmrc) [reply test/base/BaseSpec.scala:29]: :) fixed :)

File: test/services/EnglishUIBuilderSpec.scala (hunk 7 above)
Comment (cmrhmrc) [reply test/services/EnglishUIBuilderSpec.scala:948]: removed, there were many more, must of missed this one

File: test/services/WelshTextBuilderSpec.scala (hunk 8 above)
Comment (cmrhmrc) [reply test/services/WelshTextBuilderSpec.scala:57]: updated

File: test/services/WelshTextBuilderSpec.scala (hunk 9 above)
Comment (cmrhmrc) [reply test/services/WelshTextBuilderSpec.scala:95]: fixed

File: test/services/WelshUIBuilderSpec.scala (hunk 10 above)
Comment (cmrhmrc) [reply test/services/WelshUIBuilderSpec.scala:682]: unified defns

File: test/services/WelshUIBuilderSpec.scala (hunk 11 above)
Comment (cmrhmrc) [reply test/services/WelshUIBuilderSpec.scala:700]: pass now

File: test/views/components/ConfirmationPanelSpec.scala (hunk 12 above)
Comment (cmrhmrc) [reply test/views/components/ConfirmationPanelSpec.scala:56]: removed

File: test/views/components/GovukFooterSpec.scala (hunk 13 above)
Comment (cmrhmrc) [reply test/views/components/GovukFooterSpec.scala:41]: Welsh tests removed as here there is only text drawn from message files and the language functioning is tested elsewhere

File: test/views/components/H1HeadingSpec.scala (hunk 14 above)
Comment (cmrhmrc) [reply test/views/components/H1HeadingSpec.scala:46]: removed

File: test/views/components/H1HeadingWithFieldsetSpec.scala (hunk 15 above)
Comment (cmrhmrc) [reply test/views/components/H1HeadingWithFieldsetSpec.scala:44]: removed

File: test/views/components/H1HeadingWithLabelSpec.scala (hunk 16 above)
Comment (cmrhmrc) [reply test/views/components/H1HeadingWithLabelSpec.scala:41]: removed

File: test/views/components/H3HeadingSpec.scala (hunk 17 above)
Comment (cmrhmrc) [reply test/views/components/H3HeadingSpec.scala:43]: removed

File: test/views/components/InputSpec.scala (hunk 18 above)
Comment (cmrhmrc) [reply test/views/components/InputSpec.scala:120]: done

File: test/views/components/InputSpec.scala (hunk 19 above)
Comment (cmrhmrc) [reply test/views/components/InputSpec.scala:219]: done

File: test/services/EnglishUIBuilderSpec.scala (hunk 5 above)
Comment (woollett-r) [reply test/services/EnglishUIBuilderSpec.scala]: Thanks for the explanation

**Pull request merged**